tests/integration/utils.py

```diff
@@ -587,10 +587,23 @@ def update_expected_data(env_data, replace_rules):
             dep["version"], replace_rules
         )
 
-    env_data["content_manifest"]["purl"] = replace_by_rules(
-        env_data["content_manifest"]["purl"], replace_rules
-    )
-    for i, p in enumerate(env_data["content_manifest"]["dep_purls"]):
-        env_data["content_manifest"]["dep_purls"][i] = replace_by_rules(p, replace_rules)
-    for i, p in enumerate(env_data["content_manifest"]["source_purls"]):
-        env_data["content_manifest"]["source_purls"][i] = replace_by_rules(p, replace_rules)
\ No newline at end of file
+    for i, pkg in enumerate(env_data["content_manifest"]):
```

Comment:
I probably missed something in a hurry, why we cannot use 1 for cycle for all these replacements?

```
    for i, pkg in enumerate(env_data["content_manifest"]):
        purl = pkg.get("purl", "")
        env_data["content_manifest"][i]["purl"] = replace_by_rules(purl, replace_rules)

        dep_purls = pkg.get("dep_purls", "")
        for j, purl in enumerate(dep_purls):
            env_data["content_manifest"][i]["dep_purls"][j] = replace_by_rules(purl, replace_rules)
 
        source_purls = pkg.get("source_purls", "")
        for j, purl in enumerate(source_purls):
            env_data["content_manifest"][i]["source_purls"][j] = replace_by_rules(
                purl, replace_rules
            )
```